run.go

```diff
@@ -60,9 +61,24 @@ func RunWithTimestampers(ts ...timestamp.Timestamper) RunOption {
 type RunResult struct {
 	Collection     attestation.Collection
 	SignedEnvelope dsse.Envelope
+	AttestorName   string
 }
 
+// Should this be deprecated?
+// Deprecated: Use RunWithExports instead
 func Run(stepName string, signer cryptoutil.Signer, opts ...RunOption) (RunResult, error) {
+	results, err := run(stepName, signer, opts)
+	if len(results) > 1 {
+		return RunResult{}, errors.New("expected a single result, got multiple")
+	}
+	return results[0], err
+}
+
+func RunWithExports(stepName string, signer cryptoutil.Signer, opts ...RunOption) ([]RunResult, error) {
```

Comment:
Maybe I am not seeing the link attestor the right way, but to me turning `RunResult` into a slice of results is a bit confusing. It implies in my head that `Run` has been executed more than once. Instead I think I am right in saying that the link predicate is more of a way of packaging up references to the attestations? Given this, wouldn't it make more sense to return the link attestor as a field in the `RunResult` struct?

Reply:
I have no strong opinions about the naming and structure of the result. The important part is this is the boundary for signing envelopes. I'm actually not sure why RunResult has the `attestation.Collection` either. 🤔

Reply:
Yeah I was just looking at this. The `attestation.Collection` can be extracted from the `dsse.Envelope`, we could have a function attached to the RunResult type.

Reply:
I'm operating off of memory here -- but the `dsse.Envelope` stores the Collection as json bytes. Having the Collection as part of the RunResult was to allow users of the library to not have to re-parse the bytes to get the thing we just made for them.